include/net/net_ip.h

```diff
@@ -278,12 +278,8 @@ struct sockaddr_storage {
 struct net_addr {
```

Comment:
Typo in commit subject s/uninon/union/

I agree the network code looks a bit ugly because of ifdef's that are needed in the code. But if we try to limit the ram usage, then what you are proposing is no good. Why would we relax our own rules here, I think we should leave the ifdef's here.

Reply:
Myeah I am also not completely convinced on that move, though as mentioned in the commit it's a really a little bloat.